### PR TITLE
Add option to always use system file browsing window

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
Implements feature requested in #7658.

The load/save window should be really refactored further. While we refactored the worst in #6955, it's still using loads of C-style strings and related memory handling that could easily be replaced with `std::string`.

### CODE REVIEW

**Comment by marijnvdwerf** [comment]
Does this really demand yet another visible setting? The options dialog is already overpopulated as-is.

File: src/openrct2-ui/windows/LoadSave.cpp [hunk 1]
```diff
@@ -161,17 +161,85 @@ static void window_loadsave_sort_list();
...
+        case LOADSAVETYPE_LANDSCAPE:
+            return isSave ? "*.sc6" : "*.sc6;*.sv6;*.sc4;*.sv4";
+
+        case LOADSAVETYPE_SCENARIO:
+            return "*.sc6";
+
+        case LOADSAVETYPE_TRACK:
+            return isSave ? "*.td6" : "*.td6;*.td4";
+
+        case LOADSAVETYPE_IMAGE:
```
Comment (Broxzier) [inline src/openrct2-ui/windows/LoadSave.cpp]: Not very related to this PR, buw while you're at refactoring anyway, could you rename this to `LOADSAVETYE_HEIGHTMAP` and in the function above `getInitialDirectoryByType` set the heightmap directory as the default? Loading of images is not being used by anything else yet, and heightmaps could have their own default directory.

**Comment by Umdlye** [comment]
The OS-native window is called "system dialog window" in an existing string, maybe the new strings should match that.

**Comment by AaronVanGeffen** [comment]
I think "system dialog window" isn't specific enough to file browsing in the option window's context, though. Shall we compromise on using "system file browsing window" (i.e. 'system' rather than 'native')?

File: data/language/en-GB.txt [hunk 2]
```diff
@@ -3722,6 +3722,8 @@ STR_6258    :Clear (transparent)
 STR_6259    :Disabled
 STR_6260    :Show blocked tiles
 STR_6261    :Show wide paths
+STR_6262    :Always use system file browsing window
```
Comment (Gymnasiast) [inline data/language/en-GB.txt]: I'm fine with the change, but then I'd like to see the existing usage of "system dialog window" adjusted too.

Comment (AaronVanGeffen) [reply data/language/en-GB.txt]: Oh, I thought that the button in the load/save window was clear enough in context. Unless there are other usages I'm not aware of?

Comment (Umdlye) [reply data/language/en-GB.txt]: "Always use system dialog window for saving/loading" fits the options window too, by the way.

Comment (Gymnasiast) [reply data/language/en-GB.txt]: @AaronVanGeffen Possibly, but I like to use consistent terminology. I think it makes it easier for the user to understand they're referring to the same thing.

Comment (AaronVanGeffen) [reply data/language/en-GB.txt]: "Always use system dialog window for file browsing", then?

Comment (Gymnasiast) [reply data/language/en-GB.txt]: No, I meant changing it all to "system file browsing window". The existing button in the load/save window included.

Comment (AaronVanGeffen) [reply data/language/en-GB.txt]: Done.

Comment (pizza2004) [reply data/language/en-GB.txt]: Personally I think saying "System File Browser" would be better.  Browsing window has a really weird, sorta jilted cadence to it in my opinion.  Like calling a Chrome window a "Internet browsing window".

Comment (Gymnasiast) [reply data/language/en-GB.txt]: I have no objections to that, but I'll let @AaronVanGeffen decide.

Comment (AaronVanGeffen) [reply data/language/en-GB.txt]: Once more unto the breach, dear friends, once more…

**Pull request merged**